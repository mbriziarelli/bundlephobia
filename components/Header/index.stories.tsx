import React from 'react'
import Header from '.'

export default {
  component: Header,
  title: 'Header',
}

export const Default = (): React.ReactElement => <Header />
