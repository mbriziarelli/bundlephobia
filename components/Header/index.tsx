import React from 'react'
import Typography from '@material-ui/core/Typography'
import { makeStyles, createStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(() =>
  createStyles({
    logo: {
      textTransform: 'uppercase',
      fontWeight: 600,
      letterSpacing: '4px',
      userSelect: 'none',
    },
    tagline: {
      color: 'grey',
      fontWeight: 300,
      userSelect: 'none',
    },
  })
)

const Header: React.FC = () => {
  const classes = useStyles()

  return (
    <>
      <Typography
        variant="h4"
        component="h1"
        align="center"
        className={classes.logo}
      >
        bundlephobia
      </Typography>
      <Typography
        variant="h6"
        component="h2"
        gutterBottom
        align="center"
        className={classes.tagline}
      >
        find the cost of adding a npm package to your bundle
      </Typography>
    </>
  )
}

export default Header
