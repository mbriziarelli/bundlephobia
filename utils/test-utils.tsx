import React, { ReactElement } from 'react'
import { render, RenderResult, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from '../components/theme'

const AllTheProviders: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options: RenderOptions = {}
): RenderResult => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
