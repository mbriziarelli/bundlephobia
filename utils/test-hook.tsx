import React from 'react'
import { render } from './test-utils'
import { RenderResult } from '@testing-library/react'

interface Props {
  callback: () => void
}

export const TestHook: React.FC<Props> = ({ callback }) => {
  callback()
  return null
}

export const testHook = (callback: () => void): RenderResult =>
  render(<TestHook callback={callback} />)
