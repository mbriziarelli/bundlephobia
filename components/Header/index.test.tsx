/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '../../utils/test-utils'
import Header from '.'

describe('<Header /> component', () => {
  it('renders the component', () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})
