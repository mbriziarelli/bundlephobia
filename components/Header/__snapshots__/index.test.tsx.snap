// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component renders the component 1`] = `
<div>
  <h1
    class="MuiTypography-root makeStyles-logo-1 MuiTypography-h4 MuiTypography-alignCenter"
  >
    bundlephobia
  </h1>
  <h2
    class="MuiTypography-root makeStyles-tagline-2 MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-alignCenter"
  >
    find the cost of adding a npm package to your bundle
  </h2>
</div>
`;
